### DISCUSSION TITLE
Pyccel for Competitive Programming: Unimplemented Python Functionalities

### DISCUSSION THREAD

**Discussion by ceciver** [pyccel/pyccel#1332] [Ideas]
Competitive programming places a strong emphasis on the efficiency of program execution time. Participants are presented with algorithmic or mathematical problems and must develop programs that can read input and output answers within strict time constraints. To optimize their performance, many competitive programmers opt to use languages such as C++ rather than Python.

While Pyccel is a useful tool for optimizing Python code, it does not currently support all of the built-in functionalities and libraries that are commonly used in competitive programming. 

The competitive programming community is quite large. In fact, Codeforces holds around 3 contests each week, with more than 10,000 participants taking part in each contest.

This will be an ongoing discussion where I will regularly update it every time I come across a Python functionality that is commonly used in competitive programming but is not currently available in Pyceel.

This is an example of a Python 3 solution from Codeforces:

```python
import math
t=int(input())
def ncr(n,r):
    k=1
    ans=1
    while(k<=r):
        ans=ans*(n-k+1)
        ans//=k
        k+=1
    return ans%998244353
for t1 in range(t):
    l,r=map(int,input().split())
    n=math.log2(r/l)
    if(abs(int(n)-n)<1e-7):
        n=round(n)+1
    else:
        n=int(n)+1
    ans=0
    it=0
    vals=r/(2**(n-1))
    while(vals>=l):
        ans+=ncr(n-1,it)*(int(vals)-l+1)
        ans%=998244353
        it+=1
        vals=((vals*2)/3)
    
    print(n,end=" ")
    print(ans%998244353)
```
Here is a list of functionalities  that need to be implemented in Pyccel:

- [ ] input
- [ ] Inhomogeneous lists
- [ ] in operator
- [ ] not in operator
- [x] // floor division
- [ ] split
- [ ] set
- [ ] round

**Comment by EmilyBourne** [comment]
Do you have an example with inhomogeneous lists?
